packages/sdk/typescript/human-protocol-sdk/test/escrow.test.ts

```
@@ -0,0 +1,1368 @@
+          amounts,
+          finalResultsUrl,
+          finalResultsHash
+        )
+      ).rejects.toThrow(ErrorHashIsEmptyString);
+    });
+
+    test('should throw an error if escrow does not have enough balance', async () => {
+      const escrowAddress = ethers.constants.AddressZero;
+      const recipients = [
+        ethers.constants.AddressZero,
+        ethers.constants.AddressZero,
+      ];
+      const amounts = [BigNumber.from(90), BigNumber.from(20)];
+      const finalResultsUrl = VALID_URL;
+      const finalResultsHash = FAKE_HASH;
+
+      escrowClient.escrowFactoryContract.hasEscrow.mockReturnValue(true);
+      escrowClient.getBalance = vi.fn().mockReturnValue(BigNumber.from(50));
+
+      await expect(
+        escrowClient.bulkPayOut(
+          escrowAddress,
+          recipients,
+          amounts,
+          finalResultsUrl,
+          finalResultsHash
+        )
+      ).rejects.toThrow(ErrorEscrowDoesNotHaveEnoughBalance);
+    });
+
+    test('should successfully bulkPayOut escrow', async () => {
+      const escrowAddress = ethers.constants.AddressZero;
+      const recipients = [
+        ethers.constants.AddressZero,
+        ethers.constants.AddressZero,
+      ];
+      const amounts = [BigNumber.from(10), BigNumber.from(10)];
+      const finalResultsUrl = VALID_URL;
+      const finalResultsHash = FAKE_HASH;
+
+      escrowClient.escrowFactoryContract.hasEscrow.mockReturnValue(true);
+      escrowClient.getBalance = vi.fn().mockReturnValue(BigNumber.from(100));
+
+      await escrowClient.bulkPayOut(
+        escrowAddress,
+        recipients,
+        amounts,
+        finalResultsUrl,
+        finalResultsHash
+      );
+
+      expect(escrowClient.escrowContract.bulkPayOut).toHaveBeenCalledWith(
+        recipients,
+        amounts,
+        finalResultsUrl,
+        finalResultsHash,
+        DEFAULT_TX_ID
+      );
+    });
+
+    test('should throw an error if the bulkPayOut fails', async () => {
```

fix writing:
'should throw an error if bulkPayOut fails'